### PR TITLE
Tgo mapProjectReingest Test Conversion

### PR DESCRIPTION
**Pull request by acpaquette**
Converts the Tgo mapProjectReingest test into a gtest

## Description
Converts the Tgo mapProjectReingest test to our new gtest testing framework from the old make test. Also adds the ability to generate binary kernels from transfer format kernels. This is replicated from ALE. One thing that is not handled is cleaning up the binary kernels once a test is finished.

## Related Issue
#4666 

## Motivation and Context
Getting clean/stable jenks builds/tests

## How Has This Been Tested?
This is a test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/TgoCassisModuleTests.cpp [hunk 1]
```diff
@@ -1571,5 +1571,154 @@ TEST(TgoCassisModuleTests, TgoCassisColorMosaic) {
...
+  QString projCubeName = prefix.path() + "/CAS-M01-2018-05-05T23.11.48.767-RED-01029-B1.equi.cub";
+  QString mapFile = "data/tgoCassis/mapProjectedReingested/equi.map";
+  QVector<QString> cam2mapArgs = {"from=" + outputCubeName,
+                                  "to=" + projCubeName,
+                                  "map=" + mapFile,
+                                  "pixres=mpp",
+                                  "resolution=200"};
+  UserInterface cam2mapPan(CAM2MAP_XML, cam2mapArgs);
+  try {
+    cam2map(cam2mapPan);
```
Comment (amystamile-usgs) [inline isis/tests/TgoCassisModuleTests.cpp]: A couple spots where "Pan" needs to be changed to "Red"

File: isis/tests/TestUtilities.cpp [hunk 2]
```diff
@@ -250,4 +250,33 @@ namespace Isis {
...
+    QString binaryKernelList("(");
+    for (QString kernel : kernelList) {
+      FileName file(kernel);
+      QString kernelExtension = file.extension();
+      QString pathToBinaryKernel = file.path() + "/" + file.baseName() + "." + kernelExtension.replace('x', 'b');
+      FileName binaryFile(pathToBinaryKernel);
+
+      if (kernelExtension.contains("x") && !binaryFile.fileExists()) {
+        QString path = file.expanded();
+        QString command = "tobin " + path;
```
Comment (Kelvinrr) [inline isis/tests/TestUtilities.cpp:267]: Where are the executables kept? I recall these being pretty standalone and shipped separate from the c lib?

**Review by Kelvinrr — COMMENTED** [review]
just a question in line

File: isis/tests/TestUtilities.cpp (hunk 2 above)
Comment (jessemapel) [reply isis/tests/TestUtilities.cpp:267]: They are distributed with the c library. There's a handful of special case ones like ckslicer that are not, but tobin is.

**Review by jessemapel — COMMENTED** [review]
For clean-up I see two options:

1. Use fixtures or global setup and teardown to clean up the kernel files
2. Create a class to wrap the Kernel files that deletes them in its destructor

**Comment by acpaquette** [comment]
@jessemapel Resolved your comment with a test suite based fixture

**Review by jessemapel — APPROVED** [review]

**Pull request merged**